app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -235,6 +235,7 @@ def consumer_activate
       # Activation keys are userless by definition so use the internal generic user
       # Set it before calling find_activation_keys to allow communication with candlepin
       User.current = User.anonymous_admin
+      additional_set_taxonomy
```

Comment:
Here when AK is used, the `User.current` is nil when `set_taxonomy` before action is triggered, it was not triggered from user login either since this bypass authentication from core, therefore we need to load the right taxonomy here, after user login was simulated on line above. Since `param[:organization_id]` was again set to the label by `set_organization_id` callback, we need to load the right one again, therefore the `additional_set_taxonomy` does the params swapping

Reply:
Probably worth commenting directly in the code, as a message for future generation

Reply:
it's there but it's at additional_set_taxonomy method definition